src/wp-admin/network/settings.php

```diff
@@ -256,7 +256,7 @@
 					$limited_email_domains = get_site_option( 'limited_email_domains' );
 					$limited_email_domains = str_replace( ' ', "\n", $limited_email_domains );
 
-					if ( $limited_email_domains ) {
+					if ( ! is_scalar( $limited_email_domains ) ) {
 						$limited_email_domains = implode( "\n", (array) $limited_email_domains );
 					}
```

Comment:
This change is unnecessary and a false trigger from PHPStan. 

Why? [`str_replace`](https://www.php.net/manual/en/function.str-replace.php) returns either `string` or `array` for PHP 5, 7, and 8. See check here https://3v4l.org/Lfmgp. 

What about the 3rd parameter for `str_replace`? It accepts a `string` or `array`.  

- A scalar is type juggled to `string` (see the demo) in PHP 5-8
- A `string` or `array` is accepted in PHP 5-8
- Anything else results in a fatal error in PHP 5-8

Therefore, the current behavior is retained when running on PHP 8 without this commit. 

Note: There is a micro-optimization opportunity here (though not directly tied to PHP 8):

- Check `is_array()` and remove `(array)` type cast.
- When `string`, the type cast to `array` and then `implode` are skipped.